src/components/Header/Header.tsx

```diff
@@ -0,0 +1,62 @@
+import { FormEvent, useRef, useEffect } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todos: Todo[];
+  createTodo: (event: FormEvent) => Promise<void>;
+  toggleAll: () => void;
+  title: string;
+  setTitle: (value: string) => void;
+  isAdding: boolean;
+  toggleLoader: boolean;
+  isFocused: boolean;
+};
+
+export const Header: React.FC<Props> = ({
+  todos,
+  createTodo,
+  toggleAll,
+  title,
+  setTitle,
+  isAdding,
+  toggleLoader,
+  isFocused,
+}) => {
+  const newTodoField = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    newTodoField.current?.focus();
+  }, [isFocused]);
+
+  return (
+    <header className="todoapp__header">
+      {todos.length > 0 && (
+        <button
+          data-cy="ToggleAllButton"
+          type="button"
+          className={classNames(
+            'todoapp__toggle-all',
+            { active: todos.every(todo => todo.completed) },
```

Comment:
Better create a variable for `todos.every(todo => todo.completed)` and use it here. It is a good practice in such cases.